pkg/ingester/secondary_hash.go

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+//go:build !stringlabels
+
+package ingester
+
+import (
+	"github.com/prometheus/prometheus/model/labels"
+
+	ingester_client "github.com/grafana/mimir/pkg/ingester/client"
+)
+
+// this is taken from distributor.go
```

Comment:
Can we move the hashing logic to a shared package, so that we ensure we use the same? Or it's not possible because they work with different labels data types?

Reply:
Done in https://github.com/grafana/mimir/pull/6718/commits/c743a0d8594e218ca8cc1413e5d2856e0b23ba64.